medicines/doc-index-updater/src/service_bus_client/mod.rs

```diff
@@ -131,4 +144,36 @@ impl DocIndexUpdaterQueue {
         let evt = message.to_json_string()?;
         Ok(self.service_bus.send_event(evt.as_str(), duration).await?)
     }
+
+    pub async fn try_process_from_queue<T>(
+        &mut self,
+        state_manager: &StateManager,
+    ) -> Result<(), anyhow::Error>
+    where
+        T: Message,
+        RetrievedMessage<T>: ProcessRetrievalError,
+    {
+        tracing::info!("Checking for messages");
+        let retrieved_result: Result<RetrievedMessage<T>, RetrieveFromQueueError> =
+            self.receive().await;
```

Comment:
`receive` could be made private now.

Reply:
tried this, it breaks tests, so would rather not make this change just yet